src/internal/transactionSubmission.ts

```diff
@@ -70,7 +71,28 @@ export async function generateTransaction(
   args: { aptosConfig: AptosConfig } & GenerateTransactionInput,
 ): Promise<AnyRawTransaction> {
   const { aptosConfig, sender, data, options, secondarySignerAddresses, feePayerAddress } = args;
-  const payload = await generateTransactionPayload(data);
```

Comment:
we not going with 2 functions to generate abi and non-abi? I think we can have 2 different flows that one uses a Promise (abi) and one doesnt (non-abi)

Reply:
I would argue that when not using "easy mode" of Remote ABI, you should use a pregenerated ABI.  This would provide type safety that doesn't already exist.

The `generateTransaction` doesn't take an ABI, and looks it up on the fly.

The `generateTransactionWithABI` takes a pregenerated ABI, and then doesn't do any remote calls is fully synchronous, but still has type safety.  We'll need to provide an easy way to get that ABI for a transaction, which I can do now.

We could make a third `generateBcsTransaction`, that doesn't take an ABI, and doesn't provide any type checks.  Though, I recognized after changing all the tests how easy it is to screw up even without them.

Reply:
hmm yes. easy and default mode should be abi. see my comment here https://github.com/aptos-labs/aptos-ts-sdk/pull/121#pullrequestreview-1700840054

Reply:
synced offline, everything is great! Greg is great!

Reply:
We'll come back to generateBcsTransaction when someone asks for it / finds a reason specifically for it.

Will put on the next todo, generating ABI